src/components/WalletDropdown/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { useWeb3React } from '@web3-react/core'
+import { useState } from 'react'
+import styled from 'styled-components/macro'
+import { opacify } from 'theme/utils'
+
+import DefaultMenu from './DefaultMenu'
+import { LanguageMenu } from './LanguageMenu'
+import { TransactionHistoryMenu } from './TransactionMenu'
+
+const WalletWrapper = styled.div<{ height: string }>`
+  border-radius: 12px;
+  width: 320px;
+  height: ${(props) => props.height};
```

Comment:
the heights should be set in here with a prop specifying which version to use. so `authenticated ? ... : ...`

Reply:
that's the separation of concerns between the view layer (in here) and the logic (in the component)